biojava-core/src/test/java/org/biojava/nbio/core/util/CRC64ChecksumTest.java

```diff
@@ -0,0 +1,97 @@
+package org.biojava.nbio.core.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+/** 
+*  
+* @author Richard Adams
+*/
+class CRC64ChecksumTest {
+    CRC64Checksum crc64 = null;
+    private final String helloInCrc64Hex = "53C1111D27800000";
+    private final Long helloInCrc64Decimal = 6035123792567599104L;
+
+    @BeforeEach
+    void before (){
+         crc64 = new CRC64Checksum();
+    }
+
+    @Test
+    @DisplayName("Default value is 0")
+    void initialBehaviour() {
+        assertEquals(0, crc64.getValue());
+        assertEquals("0000000000000000", crc64.toString());
+    }
+
+    @RepeatedTest(10)
+    void sameInputRepeatedlyGeneratesSameOutput(){
+        crc64.update("hello");
+        assertEquals(helloInCrc64Decimal, crc64.getValue());
+        assertEquals(helloInCrc64Hex, crc64.toString());
+    }
+
+    @Test
+    void afterResettingCrcIsZero(){
+        crc64.update("hello");
+        crc64.reset();
+        assertEquals(0, crc64.getValue());
+    }
+
+    @Test
+    void addingIncrementallyIsSameAsAllAtOnce(){
+        crc64.update("h");
+        crc64.update("e");
+        crc64.update("l");
+        crc64.update("l");
+        crc64.update("o");
+        assertEquals(helloInCrc64Hex, crc64.toString());
+    }
+
+    @Test
+    void allbyteRange (){
+        byte [] testBytes = new byte [] {1,2,3,4,5};
+        crc64.update(testBytes, 0, testBytes.length);
+        String allBytesHex = crc64.toString();
+        crc64.reset();
+        for (byte b: testBytes) {
+            crc64.update(b);
+        }
+        assertEquals(allBytesHex, crc64.toString());
+    }
+
+    @Test
+    void allRangeIsSameAsAllArray (){
+        byte [] testBytes = new byte [] {1,2,3,4,5};
+        crc64.update(testBytes, 0, testBytes.length);
+        Long valueFromAllRange = crc64.getValue();
+        crc64.reset();
+        crc64.update(testBytes);
+        assertEquals(valueFromAllRange, crc64.getValue());
+    }
+
+    @Test 
+    @Disabled
+    // this doesn't work as expected
+    // and doesn't behave according to interface. The 3rd argument
+    // is treated as an index rather than a number of bytes to include
```

Comment:
It looks to me that the function is intended to take (bytes, start, end) rather than (bytes, start, length). I think this is a failure with the documentation (both non-existent javadoc and the misleadingly named parameters). It looks like it should run as expected with `crc64.update(testBytes, 2, 3);`.

Reply:
This method doesn't seem to be used anywhere internally in BioJava. The behaviour is specified in the interface    `java.util.zip.Checksum` which this class implements (javadoc reproduced below). Since this is an interface in the standard Java library  we can't change  the contract of the interface. If we assume that there is some external client code using this method,  then from their point of view fixing this  might be a breaking change, but this would be OK if it's documented and is in version 6?

```
 Updates the current checksum with the specified array of bytes.
     *
     * @param b the byte array to update the checksum with
     * @param off the start offset of the data
     * @param len the number of bytes to use for the update
```

Reply:
I think it'd be ok introducing the breaking change for 6.0.0. For sure we should document it in the CHANGELOG

Reply:
Pushed this fix and updated the changelog.
 Also, added some input validation,  following example of `java.util.zip.CRC32` implementation